src/mon/OSDMonitor.cc

```diff
@@ -1364,6 +1365,22 @@ bool OSDMonitor::should_propose(double& delay)
     return true;
   }
 
+  // propose as fast as possible if updating up_thru or pg_temp
+  // want to merge OSDMap changes as much as possible
+  if ((pending_inc.new_primary_temp.size() == 1
+      || pending_inc.new_up_thru.size() == 1)
+      && pending_inc.new_state.size() < 2) {
```

Comment:
Yeah, you are right. If lots of osd flapping and update its state. I think it predictably a lots of primary_temp and up_thru request arrived in a following short period. At this time, we do not want the first primary_temp or up_thru updated individually but expect to wait the following request arrived and update at once.